docs/dev/guidelines/server.rst

```diff
@@ -0,0 +1,148 @@
+* More important methods should be declared at the top of a class and minor methods at the end.
+
+7. Database:
+============
+
+* Write performant queries that can also deal with more 1000 objects in a reasonable time
+* Prefer one query that fetches additional data instead of many small queries, but don't overdo it. A good rule of thumb is to query not more than 3 associations at the same time.
+* Think about lazy vs. eager fetching when modeling the data types
+* Simple datatypes: immediately think about whether null should be supported as additional state or not. In most cases it is preferable to avoid null
+* Use ``Timestamp`` instead of ``Datetime``
+
+8. Comments:
+============
+
+Only if we are writing complicated algorithm and our comments will help other developers to better understand it. If we are adding a comment then our code is not self-explaining.
+
+9. Utility:
+===========
+
+Utility methods can and should be placed in a class named for specific functionality, not “miscellaneous stuff related to project”. Most of the time, our static methods belong on a related class.
+
+10. Auto configuration:
+=======================
+
+Spring Boot favors Java-based configuration. Although it is possible to use Sprint Boot with XML sources but it’s generally not recommended. You need not put all your @Configuration into a single class. The @Import annotation can be used to import additional configuration classes.
+One of the flagship features of Spring Boot is its use of Auto-configuration. This is the part of Spring Boot that makes your code simply work. It gets activated when a particular jar file is detected on the classpath. The simplest way to make use of it is to rely on the Spring Boot Starters.
+
+11. Keep your @RestController’s clean and focused:
+==================================================
+
+* RestControllers should be stateless.
+* RestControllers are by default singletons.
+* RestControllers should not execute business logic but rely on delegation.
+* RestControllers should deal with the HTTP layer of the application.
+* RestControllers should be oriented around a use-case/business-capability.
+
+Route naming conventions:
+
+* Always use kebab-case (e.g. "/exampleAssessment" → "/example-assessment").
+* The routes should follow the general structure entity > entityId > sub-entity ... (e.g. "/exercises/{exerciseId}/participations").
+* Use plural for route's entities.
+* Specify the key entity at the end of the route (e.g. "text-editor/participations/{participationId}" should be changed to "participations/{participationId}/text-editor").
+* Never specify an id that is used only for consistency and not used in the code (e.g. GET "/courses/{courseId}/exercises{exerciseId}/participations/{participationId}/submissions/{submissionId}" can be simplified to GET "/submissions/{submissionId}" because all other entities than the submission are either not needed or can be loaded without the need to specify the id).
```

Comment:
```suggestion
* Never specify an id that is used only for consistency and not used in the code (e.g. GET "/courses/{courseId}/exercises/{exerciseId}/participations/{participationId}/submissions/{submissionId}" can be simplified to GET "/submissions/{submissionId}" because all other entities than the submission are either not needed or can be loaded without the need to specify the id).
```